src/main/java/io/odpf/kafkaclient/consumer/Consumer.java

```diff
@@ -1,6 +1,55 @@
 package io.odpf.kafkaclient.consumer;
 
 import io.odpf.kafkaclient.Client;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
 
 public class Consumer extends Client {
+
+    private final KafkaConsumer<String, String> kafkaConsumer;
+    private final BufferedReader bufferedReader;
+    private String topic;
+
+    public Consumer(KafkaConsumer<String, String> kafkaConsumer) {
+        this.kafkaConsumer = kafkaConsumer;
+        bufferedReader = new BufferedReader(new InputStreamReader(System.in));
```

Comment:
this.bufferedReader

Reply:
Changed to this.bufferedReader in a newer commit